src/EditorFeatures/Core.Wpf/QuickInfo/LazyToolTip.cs

```diff
@@ -46,9 +47,25 @@ private class LazyToolTip : ForegroundThreadAffinitizedObject
                 element.ToolTipClosing += this.OnToolTipClosing;
             }
 
-            public static void AttachTo(FrameworkElement element, IThreadingContext threadingContext, Func<DisposableToolTip> createToolTip)
+            public static IDisposable AttachTo(FrameworkElement element, IThreadingContext threadingContext, Func<ReferenceCountedDisposable<DisposableToolTip>> createToolTip)
                 => new LazyToolTip(threadingContext, element, createToolTip);
 
+            public void Dispose()
+            {
+                try
+                {
+                    AssertIsForeground();
+
+                    _element.ToolTip = null;
+                    _disposableToolTip?.Dispose();
+                    _disposableToolTip = null;
+                }
+                catch (Exception ex) when (FatalError.ReportAndCatch(ex))
+                {
+                    // Avoid propagating the exception within a UI cleanup layer
+                }
+            }
+
             private void OnToolTipOpening(object sender, ToolTipEventArgs e)
```

Comment:
📝 If an exception was no the original source of a PreviewWorkspace leaking, this is the second most likely case. This type assumes that there is a strict matching of calls to `OnToolTipOpening` and `OnToolTipClosing`. If there is ever a case where `OnToolTipOpening` is called more times than `OnToolTipClosing`, the additional calls would have leaked a `PreviewWorkspace` instance.